manifests/mod/cache_disk.pp

```diff
@@ -0,0 +1,91 @@
+# @summary
+#   Installs and configures `mod_cache_disk`.
+#
+# @description
+#   This will install an configure the proper module depending on the used apache version, so
+#   - mod_cache_disk for apache version >= 2.4
+#   - mod_disk_cache for older apache versions
+# 
+# @param cache_root
+#   Defines the name of the directory on the disk to contain cache files.
+#   Default depends on the Apache version and operating system:
+#   - Debian: /var/cache/apache2/mod_cache_disk
+#   - FreeBSD: /var/cache/mod_cache_disk
+#   - Red Hat, Apache 2.4: /var/cache/httpd/proxy
+#
+# @param cache_enable
+#   Defines an array of directories to cache, the default is none
+
+# @param cache_dir_length
+#   The number of characters in subdirectory names
+#
+# @param cache_dir_levels
+#   The number of levels of subdirectories in the cache.
+#
+# @param cache_max_filesize
+#   The maximum size (in bytes) of a document to be placed in the cache
+#
+# @param cache_ignore_headers
+#   DEPRECATED Ignore request to not serve cached content to client (included for compatibility reasons to support disk_cache)
+#
+# @param configuration_file_name
+#   DEPRECATED Name of module configuration file (used for the compatibility layer for disk_cache)
+#
+# @note
+#   Apache 2.2, mod_disk_cache installed. On Apache 2.4, mod_cache_disk installed.
+#
+# @see https://httpd.apache.org/docs/2.4/mod/mod_cache_disk.html for additional documentation on version 2.4.
+#
+class apache::mod::cache_disk (
+  Optional[String] $cache_root              = undef,
```

Comment:
Are relative path acceptable?  If yes, a non-null `String` should be expected:

```suggestion
  Optional[String[1]] $cache_root              = undef,
```

otherwise:

```suggestion
  Optional[Stdlib::Absolutepath] $cache_root              = undef,
```